### PR TITLE
first

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/internet/market/controllers/AddProductInShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import internet.market.model.ShoppingCart;
+import internet.market.service.ProductService;
+import internet.market.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductInShoppingCartController extends HttpServlet {
```
Comment (AnaLanda) [inline src/main/java/internet/market/controllers/AddProductInShoppingCartController.java]:
```suggestion
public class AddProductToShoppingCartController extends HttpServlet {
```

File: src/main/java/internet/market/controllers/InjectUserDataController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import internet.market.lib.Injector;
+import internet.market.model.User;
+import internet.market.service.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class InjectUserDataController extends HttpServlet {
```
Comment (AnaLanda) [inline src/main/java/internet/market/controllers/InjectUserDataController.java]: The checklist said "Use single separate controller to inject data", so maybe you'd need to combine InjectUserDataController & InjectProductDataController

File: src/main/java/internet/market/controllers/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+
```
Comment (AnaLanda) [inline src/main/java/internet/market/controllers/ShoppingCartController.java]: Remove newline)

**Review by AnaLanda — COMMENTED** [review]
Looks good)) Left a couple minor comments.

File: src/main/java/internet/market/controllers/AddProductInShoppingCartController.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/AddProductInShoppingCartController.java]: yes, done

File: src/main/java/internet/market/controllers/ShoppingCartController.java (hunk 3 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/ShoppingCartController.java]: done

File: src/main/java/internet/market/controllers/AddProductController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/addProduct.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productName = req.getParameter("name");
+        String productPrice = req.getParameter("price");
+        if (productName.length() > 0 && productPrice.length() > 0) {
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/AddProductController.java]: checks should be done by jsp
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/internet/market/controllers/AddProductInShoppingCartController.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/internet/market/controllers/AddProductInShoppingCartController.java]: it looks not done yet

File: src/main/java/internet/market/controllers/AddProductInShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+    private ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String idUnit = req.getParameter("id");
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/AddProductInShoppingCartController.java]:
```suggestion
        String productId = req.getParameter("id");
```

File: src/main/java/internet/market/controllers/DeleteUserController.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String idUnit = req.getParameter("id");
+        Long id = Long.valueOf(idUnit);
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/DeleteUserController.java:22]: let's delete cart here as well

File: src/main/java/internet/market/controllers/InjectUserDataController.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user1 = new User("Monica Geller",
+                "turkey222", "678gkjx");
+        User user2 = new User("Ross Geller", "newyork200", "47871982735");
+        userService.create(user1);
+        userService.create(user2);
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/InjectUserDataController.java]: create carts for you newly created users as well

File: src/main/java/internet/market/controllers/ProductController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import internet.market.model.Product;
+import internet.market.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/ProductController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/internet/market/controllers/ShoppingCartController.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import internet.market.model.ShoppingCart;
+import internet.market.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/ShoppingCartController.java]:
```suggestion
public class GetShoppingCartProductsController extends HttpServlet {
```

File: src/main/java/internet/market/controllers/AddProductInShoppingCartController.java (hunk 5 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/AddProductInShoppingCartController.java]: done

File: src/main/java/internet/market/controllers/InjectUserDataController.java (hunk 7 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/InjectUserDataController.java]: done

File: src/main/java/internet/market/controllers/DeleteUserController.java (hunk 6 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/DeleteUserController.java:22]: done

File: src/main/java/internet/market/controllers/ProductController.java (hunk 8 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/ProductController.java]: done

File: src/main/java/internet/market/controllers/ShoppingCartController.java (hunk 9 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/ShoppingCartController.java]: done

File: src/main/java/internet/market/controllers/InjectUserDataController.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/InjectUserDataController.java]: okay thanks

File: src/main/java/internet/market/controllers/AddProductToShoppingCartController.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+            .getInstance(ShoppingCartService.class);
+    private ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productId = req.getParameter("id");
+        Long id = Long.valueOf(productId);
+        ShoppingCart shoppingCart = shoppingCartService.create(new ShoppingCart(USER_ID));
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/AddProductToShoppingCartController.java]: why do we create new shopping cart for user every time ? 
use existing shopping cart that you've created during registration

File: src/main/java/internet/market/controllers/RegistrationController.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        String passwordRepeat = req.getParameter("password-repeat");
+        if (password.equals(passwordRepeat)) {
+            User user = new User(name, login, password);
+            userService.create(user);
```
Comment (RomanDubovskyi) [inline src/main/java/internet/market/controllers/RegistrationController.java:36]: create shopping cart for user during registration

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -10,6 +10,86 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductInCart</servlet-name>
+        <servlet-class>internet.market.controllers.AddProductToShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addProductInCart</servlet-name>
+        <url-pattern>/cart/add</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]: use url pattern similar to the one showed in checklist 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/internet/market/controllers/AddProductToShoppingCartController.java (hunk 10 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/AddProductToShoppingCartController.java]: done

File: src/main/java/internet/market/controllers/RegistrationController.java (hunk 11 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/RegistrationController.java:36]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**